lib/block-patterns.php

```diff
@@ -302,3 +302,134 @@ function( $current_screen ) {
 		}
 	}
 );
+
+/**
+ * FIXME.
+ *
+ * @since 5.9.0
+ */
+function _register_theme_block_patterns() {
+	$default_headers = array(
+		'title'         => 'Pattern Name',
+		'description'   => 'Description',
+		'viewportWidth' => 'Viewport Width',
+		'categories'    => 'Categories',
+		'keywords'      => 'Keywords',
+		'blockTypes'    => 'Block Types',
+	);
+
+	// Register patterns for the active theme, for both parent and child theme,
+	// if applicable.
+	foreach ( wp_get_active_and_valid_themes() as $theme ) {
+		$dirpath = $theme . '/patterns/';
+		if ( file_exists( $dirpath ) ) {
+			$files = glob( $dirpath . '*.html' );
+			if ( $files ) {
+				foreach ( $files as $file ) {
+					// Parse pattern slug from file name.
+					if ( ! preg_match( '#/(?P<slug>[A-z0-9_-]+)\.html$#', $file, $matches ) ) {
+						continue; // FIXME: Consider logging notice.
+					}
+					// Example name: twentytwentytwo/query-grid-posts.
+					$pattern_name = get_stylesheet() . '/' . $matches['slug'];
+
+					$pattern_data = get_file_data( $file, $default_headers );
```

Comment:
This is an interesting idea (extracting metadata from the `.html` files). I've prepared a spin-off in case we want to pursue this direction https://github.com/WordPress/gutenberg/pull/38984

Reply:
This is really useful, thanks!